scripts/prepare_for_registration_bulk_creation.py

```diff
@@ -0,0 +1,155 @@
+        # the `csv_input` has mostly been determined, except for the `csv_parsed` which needs collaboration with
+        # the parser piece. The value of `csv_parsed` is expected to be a serializable object that can be pickled
+        # and un-pickled by celery automatically.
+        # parsing_output = {
+        #     'payload_hash': '07eef7ee-fc69-4e3a-9047-8d0e0a4cb909',
+        #     'provider_id': 'osf',
+        #     'schema_id': '60f74311ed3441000141b265',
+        #     'initiator_id': 't47ka',
+        #     'registrations': [
+        #         {
+        #             'external_id': '27093472',
+        #             'csv_raw': '1997,Ford,F250,Black',
+        #             'csv_parsed': {
+        #                 'year': 1997,
+        #                 'make': 'Ford',
+        #                 'model': 'F250',
+        #                 'color': 'Black',
+        #             },
+        #         },
+        #         {
+        #             'external_id': '98709987',
+        #             'csv_raw': '2000,Volkswagen,Golf,White',  # here goes the raw string of one row
+        #             'csv_parsed': {
+        #                 'year': 2000,
+        #                 'make': 'Volkswagen',
+        #                 'model': 'Golf',
+        #                 'color': 'White',
+        #             },
+        #         },
+        #         {
+        #             'external_id': '64510752',
+        #             'csv_raw': '2011,Subaru,WRX,Blue',  # here goes the raw string of
+        #             'csv_parsed': {
+        #                 'year': 2011,
+        #                 'make': 'Subaru',
+        #                 'model': 'WRX',
+        #                 'color': 'Blue',
+        #             },
+        #         },
+        #     ],
+        # }
+
+    logger.info('Preparing OSF database for registration bulk creation ...')
+    initiator_id = parsing_output.get('initiator_id', None)
+    if not initiator_id:
+        logger.error('Missing initiator id')  # internal error
+        return handle_internal_error()
+    initiator = OSFUser.load(initiator_id)
+    if not initiator:
+        logger.error('Initiator not found: [id={}]'.format(initiator_id))  # internal error
+        return handle_internal_error()
+
+    provider_id = parsing_output.get('provider_id', None)
+    if not provider_id:
+        logger.error('Missing provider id')  # internal error
+        return handle_internal_error()
+    provider = RegistrationProvider.objects.filter(_id=provider_id).first()
+    if not provider:
+        logger.error('Provider not found: [id={}] '.format(provider_id))  # internal error
+        return handle_internal_error()
+
+    schema_id = parsing_output.get('schema_id', None)
+    if not schema_id:
+        logger.error('Missing schema id')  # internal error
+        return handle_internal_error()
+    schema = RegistrationSchema.objects.filter(_id=schema_id).first()
```

Comment:
Same question: is it possible for two RegistrationSchemas to have the same ID?

Reply:
I forgot this: `RegistrationSchema` inherits from the legacy `ObjectIDMixin` and the `_id` is unique

```python
class ObjectIDMixin(BaseIDMixin):
    primary_identifier_name = '_id'
    _id = models.CharField(max_length=24, default=generate_object_id, unique=True, db_index=True)
```